src/averageAges.js

```diff
@@ -37,7 +49,24 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenArray = people.filter(object => object.sex === 'f');
+
+  const women = withChildren
+    ? womenArray.filter(mom => people.some(person => person
+      .mother === mom.name))
+    : womenArray;
+
+  const totalAge = women.reduce((sum, item) => {
+    const age = item.died - item.born;
+
+    return sum + age;
+  }, 0);
+
+  const totalWomen = women.length;
+
+  const averageAge = totalAge / totalWomen;
+
+  return averageAge;
```

Comment:
the same here about variables, also, you used this code twice, better create a function for it

Reply:
> 

I've deleted these variables. But I don't understand where I should use the function. There are two different functions.

Reply:
check it out, please